aggregator/aggregator.go

```diff
@@ -227,18 +272,30 @@ func (agg *aggregator) AddMetricWithPoliciesList(
 	return err
 }
 
+func (agg *aggregator) Resign() error {
+	ctx, cancel := context.WithTimeout(context.Background(), agg.resignTimeout)
+	defer cancel()
+	return agg.electionManager.Resign(ctx)
+}
+
+func (agg *aggregator) Status() RuntimeStatus {
+	return RuntimeStatus{
+		FlushStatus: agg.flushManager.Status(),
+	}
+}
+
 func (agg *aggregator) Close() error {
 	agg.Lock()
-	defer agg.Unlock()
-
 	if agg.state != aggregatorOpen {
-		return errAggregatorIsNotOpenOrClosed
+		agg.Unlock()
```

Comment:
Seems like this is missing from the final return statement?  Any reason to not just defer `agg.Unlock()` like we used  to previously?

If not we need to `agg.Unlock()` after the `close(...)` yeah?